utils/yoke/src/yoke.rs

```diff
@@ -148,6 +148,36 @@ where
     /// assert_eq!(&**yoke.get(), "hello");
     /// assert!(matches!(yoke.get(), &Cow::Borrowed(_)));
     /// ```
+    ///
+    /// Keep track of how many bytes were read from the buffer:
```

Comment:
perhaps explicitly mrention that the closure can mutate local variables? Or something that motivates this example?